src/cache-view.ts

```diff
@@ -69,7 +69,7 @@ export class ConfigurationWebview {
     if (this.dirty) {
       telemetry.logEvent("editCMakeCache", {command: "saveCMakeCacheUI"});
       await this.saveCmakeCache(this._options);
-      vscode.window.showInformationMessage(localize('cmake.cache.saved', 'CMake options have been saved.'));
+      void vscode.window.showInformationMessage(localize('cmake.cache.saved', 'CMake options have been saved.'));
```

Comment:
I don't think void is needed here and similar places in the code, i.e. before `void vscode.window.showInformationMessage`.

Reply:
required by no-floating-promises

Reply:
I see that this rule mostly applies to `vscode.window.showInformationMessage` and `vscode.window.showErrorMessage`. Trying to understand why this rule is needed? as we normally don't handle the value from these two.

Reply:
The important ones are in state.ts. Those has caused a number of problems, including many flaky test :). Adding 'void' keyword is a very cheap way to avoid these errors in the future.

Reply:
There is an option ignoreVoid you can add to this rule

Reply:
I'd like to avoid using that option. That option ignores Promise<void>. And all those update functions in state.ts return Promise<void>.

Reply:
It looks like I deleted my comment before knowing you already replied to it. Yes, I realized adding 'ignoreVoid' won't help.

Reply:
The rule originally helped in finding where we need to add await instead of just ignoring the promise, right? IMHO, we can remove the rule now, and remove all the void terms that are added to the code and only keep the awaits that are necessary. The reason is that the code needs some cleaning already, and adding so many voids looks unnecessary in my opinion.

Reply:
The thing is, it's really hard to find if anyone encounters this issue. For example, it took Bob and me months to figure out these problems in state.ts. Comparing to the debugging effort, it's much easier to write some 'void's.

Reply:
you are right. I will close this comment as resolved.